### DISCUSSION TITLE
Chaos Mesh Offline Installation - InvalidImage for chaos-mesh:v2.5.1

### DISCUSSION THREAD

**Discussion by vparikh44** [chaos-mesh/chaos-mesh#3961] [Q&A] [answered]
## Bug Report

**What version of Kubernetes are you using?**
1.23.8

**What version of Chaos Mesh are you using?**
v2.5.1

**What did you do? / Minimal Reproducible Example**
# pulled the image
docker pull ghcr.io/chaos-mesh/chaos-mesh:v2.5.1
docker pull ghcr.io/chaos-mesh/chaos-daemon:v2.5.1
docker pull ghcr.io/chaos-mesh/chaos-dashboard:v2.5.1

#saved the image
docker save ghcr.io/chaos-mesh/chaos-mesh:v2.5.1 > image-chaos-mesh.tar
docker save ghcr.io/chaos-mesh/chaos-daemon:v2.5.1 > image-chaos-daemon.tar
docker save ghcr.io/chaos-mesh/chaos-dashboard:v2.5.1 > image-chaos-dashboard.tar

#tagged the image
$CHAOS_MESH_IMAGE= "myprivate.azurecr.io/chaos-mesh/chaos-mesh:v2.5.1"
$CHAOS_DAEMON_IMAGE = "myprivate.azurecr.io/chaos-mesh/chaos-daemon:v2.5.1"
$CHAOS_DASHBOARD_IMAGE = "myprivate.chaos-mesh/chaos-dashboard:v2.5.1"
docker image tag ghcr.io/chaos-mesh/chaos-mesh:v2.5.1 $CHAOS_MESH_IMAGE
docker image tag ghcr.io/chaos-mesh/chaos-daemon:v2.5.1 $CHAOS_DAEMON_IMAGE
docker image tag ghcr.io/chaos-mesh/chaos-dashboard:v2.5.1 $CHAOS_DASHBOARD_IMAGE

# pushed the image
docker push $CHAOS_MESH_IMAGE
docker push $CHAOS_DAEMON_IMAGE
docker push $CHAOS_DASHBOARD_IMAGE

Downloaded the copy of the zip file 
https://github.com/chaos-mesh/chaos-mesh/archive/refs/tags/v2.5.1.zip

Executed the helm install after RDP into the VM hosted in the same subnet
cd chaos-mesh-2.5.1

kubectl create ns chaos-mesh
helm install chaos-mesh helm/chaos-mesh -n=chaos-mesh --set images.registry=myprivate.azurecr.io

**What did you expect to see?**
NAME                                        READY   STATUS    RESTARTS   AGE
chaos-controller-manager-69fd5c46c8-xlqpc   3/3     Running   0          2d5h
chaos-daemon-jb8xh                          1/1     Running   0          2d5h
chaos-dashboard-98c4c5f97-tx5ds             1/1     Running   0          2d5h

**What did you see instead?**

NAME                                        READY   STATUS    RESTARTS   AGE
chaos-controller-manager-69fd5c46c8-xlqpc   0/1     InvalidImageName   0          2d5h
chaos-daemon-jb8xh                          0/1     InvalidImageName   0          2d5h
chaos-dashboard-98c4c5f97-tx5ds             0/1     InvalidImageName   0          2d5h

Executing the following command
kubectl describe pods -n chaos-mesh

Message: 
failed to apply default image tag "https://myprivate.azurecr.ip/chaos-mesh/chaos-mesh:v2.5.1/chaosh-mesh/chaos-mesh:v2.5.1": couldn't parse image reference "https://myprivate.azurecr.ip/chaos-mesh/chaos-mesh:v2.5.1/chaosh-mesh/chaos-mesh:v2.5.1" : invalid reference format

failed to apply default image tag "https://myprivate.azurecr.ip/chaos-mesh/chaos-mesh:v2.5.1/chaosh-mesh/chaos-daemon:v2.5.1": couldn't parse image reference "https://myprivate.azurecr.ip/chaos-mesh/chaos-mesh:v2.5.1/chaosh-mesh/chaos-daemon:v2.5.1" : invalid reference format

failed to apply default image tag "https://myprivate.azurecr.ip/chaos-mesh/chaos-mesh:v2.5.1/chaosh-mesh/chaos-dashboard:v2.5.1": couldn't parse image reference "https://myprivate.azurecr.ip/chaos-mesh/chaos-mesh:v2.5.1/chaosh-mesh/chaos-dashboard:v2.5.1" : invalid reference format

**Output of chaosctl**

**Comment by STRRL** [comment]
It seems that required images (like `myprivate.azurecr.ip/chaos-mesh/chaos-mesh:v2.5.1`) do not exist on your kubernetes worker node.

Please make sure that the images you prepared on the worker node are same as  image in the Pods definitions.

**Comment by vparikh44** [comment]
@STRRL , thank you for your response.  yes indeed that was the issue.  Also, I had to update the yaml file to make the required changes in the file.

This is resolved, now.

**Comment by vparikh44** [comment, marked as answer]
@STRRL , thank you for your response.  yes indeed that was the issue.  Also, I had to update the file to make the required changes in the file.

This is resolved, now.
